distributed/tests/test_worker_client.py

```diff
@@ -297,3 +297,17 @@ def f():
 
     result = yield c.submit(f)
     assert result
+
+
+@gen_cluster(client=True)
+def test_submit_different_names(c, s, a, b):
+    # https://github.com/dask/distributed/issues/2058
+    da = pytest.importorskip('dask.array')
+    c = yield Client('localhost:' + s.address.split(":")[-1], loop=s.loop,
+                     asynchronous=True)
+    with c:
+        X = c.persist(da.random.uniform(size=(100, 10), chunks=50))
+        yield wait(X)
+
+        fut = yield c.submit(lambda x: x.sum().compute(), X)
+        assert fut > 0
```

Comment:
I recommend avoiding the use of the normal context manager in async tests.  If you were to use `async with c` that would work well, but we can't use that within tests that have to be within python 2 friendly files.  Instead I recommend using try/finally.  

Also, if you're not going to use the client created in the gen_cluster decorator then you probably want the following:

```python
@gen_cluster()
def test_foo(s, a, b):
    c = yield Client(...)

    try:
        ...
    finally:
        yield c.close()
```

There are also some examples of this in test_client.py

Reply:
OK, thanks. I was just using the context manager so that it was cleaned up.

Reply:
Understood, but the context manager is synchronous API.  There is no way for us to `yield c.close()` within the normal context mananger.  I think that it just starts the close coroutine, but doesn't wait for it.  You'll either need to use `async with` or explicitly call `c.close()` to be sure that it closes up.